tools/types.ts

```diff
@@ -1,6 +1,22 @@
+type RemoveIndex<T extends Record<string, unknown>> = Pick<T, KnownKeys<T>>;
+type KnownKeys<T> = {
```

Comment:
I don't have a problem with what the script is doing, but these types look pretty hard to maintain :/

Reply:
this is standard mapped type if one wants get rid of index signatures. Ideally we should have a package that implements this utilities. It's definitely overkill for this migration script but because `CompilerOptions` is implemented how it is, this is radically easier effort than having whole compiler options written by hand and keep synced with actuall TS version

Reply:
I'll add some JSDoc for explanation what it does